modules/planning/math/piecewise_jerk/piecewise_jerk_problem.cc

```diff
@@ -104,6 +104,19 @@ bool PiecewiseJerkProblem::OptimizeWithOsqp(
   return true;
 }
 
+void PiecewiseJerkProblem::SetZeroOrderReference(std::vector<double> x_ref) {
+  if (x_ref.size() == num_of_knots_) {
+    x_ref_ = std::move(x_ref);
```

Comment:
A copy is always created. Just pass `const std::vector<double>&` and assign it to `x_ref_`. You saved a copy when `x_ref.size() != num_of_knots_`.

Reply:
Copy is not necessary if calling the function this way: SetZeroOrderReference(std::move(some_vector)). The caller can decide if some_vector will not be used any longer. In this way, some_vector will be moved to x_ref_ without copying.